Display actual keybindings in the UI by reading the keybinding.json

hassoncs opened the pull request:
WIP

This commit introduces a new utility for reading and displaying platform-specific keybindings within the VS Code extension since VSC does not provide an API for it! 😭 

Key changes include:
- **New `keybindings.ts` utility**: Provides functions to retrieve current keybinding labels for commands, considering user overrides and default keybindings. It also handles platform-specific formatting (e.g., Cmd on macOS, Ctrl on Windows/Linux) and chord keybindings.
- **New `vscode-config.ts` utility**: Handles reading user `keybindings.json` files, supporting various VS Code forks (VSCodium, Cursor, Windsurf) and portable installations. It parses JSON5 to allow comments and trailing commas.
- **Webview integration**: The `GhostServiceSettings` component now uses a new `useKeybindings` hook to dynamically fetch and display the correct keybinding for the "Quick Inline Task" in its description. This ensures users see the shortcut relevant to their OS and custom settings.
- **Terminal Welcome Service update**: The `TerminalWelcomeService` now uses the new keybinding utility to display the correct shortcut for generating terminal commands.
- **Message passing**: Added new `getKeybindings` and `keybindings` message types for communication between the webview and the extension backend to facilitate keybinding retrieval.
- **Internationalization updates**: Modified i18n files across multiple languages to support dynamic keybinding interpolation in relevant strings.
- **Tests**: Comprehensive unit and integration tests for the new keybinding and vscode-config utilities, covering various platforms, file formats, and edge cases.

This feature enhances user experience by providing accurate and context-aware keybinding information directly within the UI, reducing confusion and improving discoverability of shortcuts.## Context

<!-- Brief description of WHAT you’re doing and WHY. -->

## Implementation

<!--

Some description of HOW you achieved it. Perhaps give a high level description of the program flow. Did you need to refactor something? What tradeoffs did you take? Are there things in here which you’d particularly like people to pay close attention to?

-->

## Screenshots

| before | after |
| ------ | ----- |
|        |       |

## How to Test

<!--

A straightforward scenario of how to test your changes will help reviewers that are not familiar with the part of the code that you are changing but want to see it in action. This section can include a description or step-by-step instructions of how to get to the state of v2 that your change affects.

A "How To Test" section can look something like this:

- Sign in with a user with tracks
- Activate `show_awesome_cat_gifs` feature (add `?feature.show_awesome_cat_gifs=1` to your URL)
- You should see a GIF with cats dancing

-->

## Get in Touch

<!-- We'd love to have a way to chat with you about your changes if necessary. If you're in the [Kilo Code Discord](https://kilocode.ai/discord), please share your handle here. -->

Commit:
move settings around

Commit:
remove now unused translation key

Commit:
change texts a bit

markijbema commented on src/utils/keybindings.ts:
```diff
@@ -0,0 +1,172 @@
+	const keybindings = packageJson.contributes?.keybindings as KeybindingEntry[] | undefined
+	const binding = keybindings?.find((kb) => kb.command === commandId)
+	if (!binding) {
+		throw new Error(`Command '${commandId}' not found in package.json keybindings`)
```
given the purpose of this functionality (showing the user a key), I would say return null or 'not found' would be better. If we want to detect it we can still send an event to posthog, but it doesnt seem important enough to throw na exception right?

markijbema commented on src/utils/keybindings.ts:
```diff
@@ -0,0 +1,172 @@
+	})
+
+	if (formattedChords.length > 1) {
+		return formattedChords.join(", ")
```
join works fine on an array of one or zero elements as well right?

markijbema commented on src/shared/WebviewMessage.ts:
```diff
@@ -335,6 +336,7 @@ export interface WebviewMessage {
 	filters?: { type?: string; search?: string; tags?: string[] }
 	settings?: any
 	url?: string // For openExternal
+	commandIds?: string[] // kilocode_change: For getKeybindings
```
for merging its easier to keep all kilocode_changes next to eachother

markijbema reviewed (approved):
Because this is mostly 'view' and not actually invoked in any logic i think it is fine. But I think it will be better to have it be robust and not crash, because the use also doesnt really is worth to crash for

Commit:
translate

Commit:
hide provider behind advanced settings

Commit:
add test

Commit:
'chore: WIP'

Commit:
Merge branch 'mark/autocomplete-settings' into display-proper-keyboard-shortcuts

Commit:
feat(ui): display keybindings in settings

Introduces the `useKeybindings` hook to fetch and manage keyboard shortcut information.
This hook is integrated into the `GhostServiceSettings` component to dynamically display
keybindings next to their respective options. Internationalization files have been
updated to support these dynamic keybindings and provide a fallback for missing ones.

Commit:
Merge remote-tracking branch 'origin/main' into display-proper-keyboard-shortcuts
